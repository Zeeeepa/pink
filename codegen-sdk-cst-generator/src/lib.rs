mod generator;
pub(crate) mod parser;
use std::error::Error;

pub fn generate_cst(source: &str, language: &str) -> Result<(), Box<dyn Error>> {
    let node_types = parser::parse_node_types(source)?;
    let cst = generator::generate_cst(&node_types)?;
    let out_dir = std::env::var("OUT_DIR").unwrap();
    let out_file = format!("{}/{}.rs", out_dir, language);
    std::fs::write(out_file, cst)?;
    Ok(())
}
