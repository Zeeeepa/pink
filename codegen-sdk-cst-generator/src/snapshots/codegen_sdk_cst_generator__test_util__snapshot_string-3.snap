---
source: codegen-sdk-cst-generator/src/lib.rs
expression: "StringDebug { string: formatted }"
---
# [rkyv (omit_bounds)] pub test_field : Vec < TestType >
